ipftrace/ipftrace.py

```diff
@@ -100,20 +100,42 @@ def parse_data(self, data):
 
 class IPFTracer:
     def __init__(
-        self, iv, saddr, daddr, proto, sport, dport, module, regex, length, manifest
+        self, iv, saddr, daddr, proto, sport, dport, module, regex, length,
+        manifest, verbose,
     ):
+        self._verbose = verbose
         self._opts = self._build_opts(iv, saddr, daddr, proto, sport, dport)
-        self._functions = self._read_manifest(manifest)
+        self._functions = self._read_functions(manifest)
         self._module = self._load_module(module)
         self._regex = regex
         self._length = length
         self._egress_functions = []
         self._flows = {}
 
+    def _read_functions(self, manifest):
+        funcs = self._read_manifest(manifest)
+        available_funcs = self._read_available_filter_functions()
+        filtered_funcs = []
+        for f in funcs:
+            if f["name"] in available_funcs:
+                filtered_funcs.append(f)
+            else:
+                if self._verbose:
+                    print(f'Function {f["name"]} is not traceable. Skip.')
+        return filtered_funcs
+
     def _read_manifest(self, manifest):
         with open(manifest) as f:
             return yaml.load(f, Loader=yaml.FullLoader)["functions"]
 
+    def _read_available_filter_functions(self):
+        path = "/sys/kernel/debug/tracing/available_filter_functions"
+        available = []
```

Comment:
Nit: `X in List` has O(N) time complexity. It would be nice to use Set instead of it.